### PR TITLE
Strip whitespace out of feature/actor/group values posted by UI

### PR DESCRIPTION
**Pull request by KevinBongart**
Fix https://github.com/jnunemaker/flipper/issues/204

By investigating the initial issue, I was able to identify several areas impacted by it.

1. Using the UI to create two features with similar names but different whitespace:

```ruby
> $flipper.features
=> #<Set: {#<Flipper::Feature:70136261075520 name="   whitespace", state=:conditional, enabled_gate_names=[:actor], adapter=:memoizable>,
 #<Flipper::Feature:70136261217520 name="   whitespace   ", state=:conditional, enabled_gate_names=[:actor], adapter=:memoizable>}>
```

2. Still via the UI, adding an actor to the second one seemed to have actually added it to the first one.

3. Both links in the UI leads to the same feature:

screen shot 2016-11-23 at 11 01 10 am

4. Copy-pasting actor values from one Flipper UI to another adds some whitespace which ends up in Flipper:

```ruby
> $flipper["   whitespace"].actors_value
=> #<Set: {"    test@blueapron.com\t"}>
```

This PR fixes all of these issues above by applying [`String#strip`](http://ruby-doc.org/core-2.3.3/String.html#method-i-strip) to the params send from the front-end to the back-end of Flipper UI.

cc @jnunemaker & @AlexWheeler for review/feedback, and @susanblueapron for finding the bug in the first place

### CODE REVIEW

File: lib/flipper/ui/actions/actors_gate.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ def get
         def post
           feature_name = Rack::Utils.unescape(request.path.split('/')[-2])
           feature = flipper[feature_name.to_sym]
-          value = params["value"]
+          value = params["value"].strip
 
           if Util.blank?(value)
```
Comment (midu) [inline lib/flipper/ui/actions/actors_gate.rb]: looks like the code expects a possible `nil` value here, which means that your change on line `27` will raise an `undefined method for nil nilclass`

maybe `strip` inside the if block?

File: lib/flipper/ui/actions/features.rb [hunk 2]
```diff
@@ -34,7 +34,7 @@ def post
             halt view_response(:feature_creation_disabled)
           end
 
-          value = params["value"]
+          value = params["value"].strip
```
Comment (midu) [inline lib/flipper/ui/actions/features.rb]: same as above

File: spec/flipper/ui/actions/actors_gate_spec.rb [hunk 3]
```diff
@@ -46,13 +48,23 @@
...
     end
 
     context "disabling an actor" do
+      let(:value) { "User:6" }
+
       before do
         flipper[:search].enable_actor Flipper::UI::Actor.new("User:6")
         post "features/search/actors",
-          {"value" => "User:6", "operation" => "disable", "authenticity_token" => token},
+          {"value" => value, "operation" => "disable", "authenticity_token" => token},
```
Comment (midu) [inline spec/flipper/ui/actions/actors_gate_spec.rb]: maybe add a test where you don't pass a `value`? seems like that's missing here

**Comment by KevinBongart** [comment]
@midu Added more tests related to empty values (not exactly the same as a param hash missing the `"value"` key but close enough). Not sure what's best between:

1. `value = params["value"].to_s.strip`
2. `value = params["value"]&.strip` (only compatible with Ruby 2.3+ I believe?
3. `value = params["value"] && params["value"].strip

**Comment by jnunemaker** [comment]
My preference is to_s.strip. It works with older versions of ruby and flipper doesn't care about nil uniquely since nil/blank are treated the same.
